sklearn/datasets/tests/test_common.py

```diff
@@ -41,3 +62,49 @@ def check_as_frame(bunch, fetch_func_partial,
         assert np.all(frame_bunch.data.dtypes == expected_data_dtype)
     if expected_target_dtype is not None:
         assert np.all(frame_bunch.target.dtypes == expected_target_dtype)
+
+
+def _has_network():
+    return bool(os.environ.get("SKLEARN_SKIP_NETWORK_TESTS", False))
```

Comment:
related: https://github.com/scikit-learn/scikit-learn/pull/17553 by @thomasjpfan

Reply:
I'm not sure if you wanna merge as is or take that PR into account.

Reply:
Ideally, I would have like to merge it quickly to avoid redundant tests in the `fetch_*` PRs that are coming. I also want the PR of @thomasjpfan to get in because this is really useful but I think that the fix will be easy to handle even once merged.